ivadomed/loader/consolidation.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+import typing
+from typing import List, Tuple
+
+from ivadomed.keywords import ModelParamsKW, DataloaderKW
+from ivadomed.loader.dataset_group import DatasetGroup
+
+if typing.TYPE_CHECKING:
+    from ivadomed.loader.all_dataset_group import AllDatasetGroups
+from ivadomed.loader.generalized_loader_configuration import GeneralizedLoaderConfiguration
+from ivadomed.loader.mri2d_segmentation_dataset import MRI2DSegmentationDataset
+
+from loguru import logger
+
+
+class ConsolidatedDataset(MRI2DSegmentationDataset):
```

Comment:
IIUC the hierarchy: `FilesDataset` [2D/3D] < `DatasetGroup` < `AllDatasetGroup` < `ConsolidatedDataset`

On reviewing the code, I guess: 
1. The `ConsolidatedDataset` class should also inherit `MRI3DSubVolumeSegmentationDataset` to address the 3D case. So, while calling their constructors, we could do something along the lines of:
```
class ConsolidatedDataset(MRI2DSegmentationDataset, MRI3DSubVolumeSegmentationDataset):
    def __init__(
        self, filename_pairs: list, config: GeneralizedLoaderConfiguration
    ):
        ...

        if (ModelParamsKW.IS_2D not in config.model_params and not config.model_params.get(ModelParamsKW.IS_2D)):
            # call constructor of MRI3DSubVolumeSegmentationDataset 
            MRI3DSubVolumeSegmentationDataset.__init__(
                ...
            )
        else:
            # call constructor of MRI2DSegmentationDataset
            MRI2DSegmentationDataset.__init__(
                ...
            )

```
2. Moreover, I would rather suggest merging `ConsolidatedDataset` into `AllDatasetGroup` based on their functionality. This is just a suggestion as I believe creating a separate class for the functions we have in `ConsolidatedDataset` is an overkill.

Reply:
> 1. The `ConsolidatedDataset` class should also inherit `MRI3DSubVolumeSegmentationDataset` to address the 3D case.

Yes I agree, thanks @kanishk16 for pointing this out.

I just re-tested this PR in light of this [comment](https://github.com/ivadomed/ivadomed/pull/1224#discussion_r1055622479) in PR #1224 and the current implementation is only working in  2D, which I missed before. So yes , `ConsolidatedDataset` absolutely needs to take account of both cases.

In the same way, the call for consolidated dataset [here](https://github.com/ivadomed/ivadomed/blob/11c67dc5e5a69b3922c9e7e7d79a00da8413c7e9/ivadomed/main.py#L684) will need to take into account that `load_filenames` is inside the constructor in 3D and is outside the constructor in 2D. This difference can be seen in the current bids pathway in [loader.py](https://github.com/ivadomed/ivadomed/blob/11c67dc5e5a69b3922c9e7e7d79a00da8413c7e9/ivadomed/loader/loader.py#L78) where `load_filenames` is only explicitly called for 2D.
(I don't know where this difference comes from, it may be possible to harmonize 2D and 3D?)